test/tools/unified-spec-runner/entities.ts

```diff
@@ -149,13 +162,23 @@ export class UnifiedMongoClient extends MongoClient {
   } as const;
 
   constructor(uri: string, description: ClientEntity) {
+    const logCollector = new Transform({
+      objectMode: true,
+      transform(log: LogMessage, _: string, next: (e: Error | null, log: LogMessage) => void) {
+        next(null, log);
+      }
+    });
```

Comment:
I was curious about the decision to use a transform stream and manually listen for data events on it.  What was the rationale behind this design?  

Why not make a log sink?

```typescript

class UnifiedLogCollector extends Writable {
  collectedEvents = [];

  _write(chunk) { // override syntax might not be right here but it's easy to do
    if (<conditions relevant to collecting logs>) {
      this.collectedEvents.push(chunk);
    }
  }
}

// and then in the UTR

export class UnifiedMongoClient extends MongoClient {
  // ...
  constructor() {
    this.logCollector = new UnifiedLogCollector();
  }

```

The new methods added to `UnifiedMongoClient` would then just become `getter`s that proxy onto methods on the `UnifiedLogCollector`.  This better encapsulates all the logging-related UTR logic into the `UnifiedLogCollector` (it doesn't pollute the `UnifiedMongoClient`) and we don't have to worry about manually consuming data from the stream.

Reply:
I don't feel strongly about using a Nodejs stream or just implementing the interface.  But I don't like using a transform stream and listening to the data events, it feels like the wrong choice conceptually.  We're not passing the logs along anywhere here - so a transform or a passthrough stream is the wrong choice because we don't need the readable capability of duplex streams.

Reply:
We can switch to the minimal interface when the code allows for it in NODE-4813. PassThrough + data even will be easy to replace since the real logic is inside pushLogMessage and it's a way to get the test runner past the current stream instanceof check to unlock debugging when the driver starts calling `.write`

Reply:
I think Bailey is right here and we should use a proper writable stream that just accumulates the logs for later access. It's not much more work and encapsulates the log filtering well.